Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -750,13 +750,33 @@ extension HTTPClient {
             return self.promise.futureResult
         }
 
+        #if swift(>=5.7)
         /// Waits for execution of this request to complete.
         ///
-        /// - returns: The value of the `EventLoopFuture` when it completes.
-        /// - throws: The error value of the `EventLoopFuture` if it errors.
+        /// - returns: The value of  ``futureResult`` when it completes.
+        /// - throws: The error value of ``futureResult`` if it errors.
+        @available(*, noasync, message: "wait() can block indefinitely, prefer get()", renamed: "get()")
```

Comment:
You should be able to just `#if` around the available attribute and avoid the duplicate impl, if you wanted to

Reply:
This is a Swift 5.8 feature: https://github.com/apple/swift-evolution/blob/main/proposals/0367-conditional-attributes.md